src/database/memory.rs

```diff
@@ -486,15 +486,23 @@ impl ConfigurableDatabase for MemoryDatabase {
 /// don't have `test` set.
 macro_rules! populate_test_db {
     ($db:expr, $tx_meta:expr, $current_height:expr$(,)?) => {{
+        $crate::populate_test_db!($db, $tx_meta, $current_height, (@coinbase false))
+    }};
+    ($db:expr, $tx_meta:expr, $current_height:expr, (@coinbase $is_coinbase:expr)$(,)?) => {{
         use std::str::FromStr;
         use $crate::database::BatchOperations;
         let mut db = $db;
         let tx_meta = $tx_meta;
         let current_height: Option<u32> = $current_height;
+        let input = if $is_coinbase {
+            vec![$crate::bitcoin::TxIn::default()]
+        } else {
+            vec![]
+        };
```

Comment:
Can't we just have a pure function here than a macro?? 😭😭😭

Life is already difficult enough.. 😂😂

Reply:
I just added a parameter to the `populate_test_db` macro to create coinbase inputs, should I duplicate the whole populate test db code just for using a function? ...

Reply:
I wonder if there's a way to do this recursively.. I will give it a shot and see..